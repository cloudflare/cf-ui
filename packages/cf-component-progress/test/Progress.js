const React = require('react');
const assertEqualJSX = require('assert-equal-jsx');
const Progress = require('../src/Progress');
const Link = require('cf-component-link');

describe('Progress', function() {
  it('should render', function() {
    assertEqualJSX(
      <Progress
        active="foo"
        onChange={() => {}}
        steps={[
          { id: 'foo', label: 'Foo', disabled: false },
          { id: 'bar', label: 'Bar', disabled: false },
          { id: 'baz', label: 'Baz', disabled: true }
        ]}
      />,
      // should equal:
      <div className="cf-progress">
        <progress className="cf-progress__bar" max={3} value={1} />
        <ol className="cf-progress__items">
          <li
            className="cf-progress__item cf-progress__item--active"
            style={{ width: '33.3333%' }}
          >
            <Link onClick={() => {}}>Foo</Link>
          </li>
          <li className="cf-progress__item" style={{ width: '33.3333%' }}>
            <Link onClick={() => {}}>Bar</Link>
          </li>
          <li
            className="cf-progress__item cf-progress__item--disabled"
            style={{ width: '33.3333%' }}
          >
            <Link onClick={() => {}} disabled>Baz</Link>
          </li>
        </ol>
      </div>
    );
  });
});
