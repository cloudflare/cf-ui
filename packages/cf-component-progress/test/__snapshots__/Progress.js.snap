// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="cf-progress"
>
  <progress
    className="cf-progress__bar"
    max={3}
    value={1}
  />
  <ol
    className="cf-progress__items"
  >
    <li
      className="cf-progress__item cf-progress__item--active"
      style={
        Object {
          "width": "33.3333%",
        }
      }
    >
      <a
        className="cf-link"
        href="#!"
        onClick={[Function]}
        role="button"
      >
        Foo
      </a>
    </li>
    <li
      className="cf-progress__item"
      style={
        Object {
          "width": "33.3333%",
        }
      }
    >
      <a
        className="cf-link"
        href="#!"
        onClick={[Function]}
        role="button"
      >
        Bar
      </a>
    </li>
    <li
      className="cf-progress__item cf-progress__item--disabled"
      style={
        Object {
          "width": "33.3333%",
        }
      }
    >
      <a
        className="cf-link cf-link--disabled"
        disabled={true}
        href="#!"
        onClick={[Function]}
        role="button"
      >
        Baz
      </a>
    </li>
  </ol>
</div>
`;
